addons/hitreactions/XEH_postInit.sqf

```diff
@@ -0,0 +1,35 @@
+#include "script_component.hpp"
+
+GVAR(armComponents) = [
+    "leftshoulder",
+    "rightshoulder",
+    "lefthand",
+    "leftforearm",
+    "leftarmroll",
+    "rightforearm",
+    "rightarmroll",
+    "righthand",
+    "rightarm"
+];
+
+["ace_firedNonPlayer", {
+    params ["", "", "", "", "", "", "_projectile"];
+    private _willDropGun = (random 1) > GVAR(weaponDropChanceGunHit);
+    private _willDropArm = (random 1) > GVAR(weaponDropChanceArmHit);
+    if (!(_willDropGun && _willDropArm)) exitWith {};
```

Comment:
```suggestion
    private _roll = random 1;
    private _willDrop = _roll < GVAR(weaponDropChanceGunHit) || {_roll < GVAR(weaponDropChanceArmHit)};
    if (!_willDrop) exitWith {};
```
Was this working before? Am I going crazy?

I'm not a fan of the concept but adding the EH on fired is the best way to do it.

Reply:
you're not going crazy, was me being stupid.

Reply:
lazy eval there is unnecessary, will cost more performance than the test